core/src/main/java/software/amazon/awssdk/runtime/StandardMemberCopier.java

```diff
@@ -74,6 +74,19 @@ public static ByteBuffer copy(ByteBuffer bb) {
         if (bb == null) {
             return null;
         }
-        return bb.duplicate();
+
+        ByteBuffer copy = bb.isDirect() ? ByteBuffer.allocateDirect(bb.capacity()) : ByteBuffer.allocate(bb.capacity());
```

Comment:
There's sometimes some thought that goes into deciding whether to use direct or non-direct bytebuffers. I'm just relying on the caller to make that decision in this case since I'm not sure if I can please everyone myself. Do you think it would be better to just stick with one in our case?